### PR TITLE
Rewrote all uses of mysql_* methods in LyricWiki code to use the MediaWi...

### PR DESCRIPTION
**Pull request by SeanColombo**
...ki database classes instead. LYR-229

All the spots that need to be fixed:
extras.php - rewritten... it was a utility function (lw_simpleQuery) used by Special:ArtistRedirects on the main page where it links to paginated results: http://lyrics.sean.wikia-dev.com/Special:ArtistRedirects?action=purge

server.php - rewritten & confirmed that I didn't break the main functionality of the page: http://lyrics.sean.wikia-dev.com/api.php?action=lyrics&song=Guitar&artist=Cake The code being rewritten was for api request-tracking that we don't use right now anyway.

Special_LinksToRedirect.php - rewritten & confirmed to work: http://lyrics.sean.wikia-dev.com/Special:Linkstoredirects?cache=clear

Special_MobileSearch.php - rewritten & confirmed to work: http://lyrics.wikia.com/Special:MobileSearches?cache=clear

Special_Soapfailures.php - rewritten & confirmed to work: http://lyrics.sean.wikia-dev.com/Special:Soapfailures?cache=clear

### CODE REVIEW

File: extensions/3rdparty/LyricWiki/server.php [hunk 1]
```diff
@@ -2509,8 +2510,13 @@ function requestStarted($funcName, $requestData){
 function requestFinished($id){
 	if( !wfReadOnly() ) {
 		if(defined('TRACK_REQUEST_RUNTIMES') && TRACK_REQUEST_RUNTIMES){
-			$db = lw_connect();
-			mysql_query("DELETE FROM apiRequests WHERE id=$id", $db);
+			$dbw = &wfGetDB(DB_MASTER);
```
Comment (macbre) [inline extensions/3rdparty/LyricWiki/server.php]: `&` can be omitted

File: extensions/3rdparty/LyricWiki/server.php [hunk 2]
```diff
@@ -2488,12 +2488,13 @@ function requestStarted($funcName, $requestData){
...
+			$dbw = wfGetDB( DB_MASTER );
 			$requestData = str_replace("'", "[&apos;]", $requestData);
 			$queryString = "INSERT INTO apiRequests (requestedThrough, requestedFunction, requestData, requestTime)";
-			$queryString.= " VALUES ('$REQUEST_TYPE', '".mysql_real_escape_string($funcName, $db)."', '".mysql_real_escape_string($requestData, $db)."', NOW())";
-			if( mysql_query($queryString, $db ) ){
-				$retVal = mysql_insert_id( $db );
+			$queryString.= " VALUES ('$REQUEST_TYPE', '".$dbw->strencode($funcName)."', '".$dbw->strencode($requestData)."', NOW())";
+			if($dbw->query($queryString)){
+				$dbw->commit();
+				$retVal = $dbw->insertId();
```
Comment (macbre) [inline extensions/3rdparty/LyricWiki/server.php]: `insertId` should be called before the `commit`.

File: extensions/3rdparty/LyricWiki/extras.php [hunk 3]
```diff
@@ -81,12 +81,11 @@ function sandboxParse($wikiText)
...
 	$retVal = "";
-	$db = wfGetDB(DB_SLAVE)->getProperty('mConn');
 
-	// TODO: use Database class instead
-	if($result = mysql_query($queryString,$db)){
-		if(($numRows = mysql_num_rows($result)) && ($numRows > 0)){
-			$row = mysql_fetch_row($result);
+	$dbr = wfGetDB(DB_SLAVE);
+	if($result = $dbr->query( $queryString, __METHOD__ )){
+		if(($numRows = $dbr->numRows($result)) && ($numRows > 0)){
```
Comment (macbre) [inline extensions/3rdparty/LyricWiki/extras.php]: Is `$numRows` actually needed here?

File: extensions/3rdparty/LyricWiki/extras.php [hunk 4]
```diff
@@ -81,12 +81,11 @@ function sandboxParse($wikiText)
...
 function lw_simpleQuery($queryString){
 	$retVal = "";
-	$db = wfGetDB(DB_SLAVE)->getProperty('mConn');
 
-	// TODO: use Database class instead
-	if($result = mysql_query($queryString,$db)){
-		if(($numRows = mysql_num_rows($result)) && ($numRows > 0)){
-			$row = mysql_fetch_row($result);
+	$dbr = wfGetDB(DB_SLAVE);
+	if($result = $dbr->query( $queryString, __METHOD__ )){
```
Comment (macbre) [inline extensions/3rdparty/LyricWiki/extras.php]: Does setting `LIMIT 1` for this query make sense (we return the first row only below)?

File: extensions/3rdparty/LyricWiki/Special_Soapfailures.php [hunk 5]
```diff
@@ -180,23 +180,25 @@ function execute(){
...
-						for($cnt=0; $cnt<$numRows; $cnt++){
-							$row = array();
-							$row['artist'] = mysql_result($result, $cnt, "request_artist");
-							$row['song'] = mysql_result($result, $cnt, "request_song");
-							$row['numRequests'] = mysql_result($result, $cnt, "numRequests");
-							$row['lookedFor'] = mysql_result($result, $cnt, "lookedFor");
-							$row['lookedFor'] = formatLookedFor($row['lookedFor']);
-							$data[] = $row;
-						}
+					for($cnt=0; $cnt < $dbr->numRows($result); $cnt++){
```
Comment (macbre) [inline extensions/3rdparty/LyricWiki/Special_Soapfailures.php]: Please use this instead:

```php
while ($row = $dbr->fetchRow( $result )) {
  // ...
}
```

File: extensions/3rdparty/LyricWiki/Special_MobileSearches.php [hunk 6]
```diff
@@ -153,23 +153,25 @@ function execute(){
...
-							$row['song'] = mysql_result($result, $cnt, "request_song");
-							$row['numRequests'] = mysql_result($result, $cnt, "numRequests");
-							$row['lookedFor'] = mysql_result($result, $cnt, "lookedFor");
-							$row['lookedFor'] = formatLookedFor($row['lookedFor']);
-							$data[] = $row;
-						}
+					for($cnt=0; $cnt < $dbr->numRows($result); $cnt++){
+						$row = $dbr->fetchRow( $result );
+						$row['lookedFor'] = formatLookedFor($row['lookedFor']);
+						$data[] = $row;
```
Comment (macbre) [inline extensions/3rdparty/LyricWiki/Special_MobileSearches.php]: Please use this instead:

```php
while ($row = $dbr->fetchRow( $result )) {
  // ...
}
```

File: extensions/3rdparty/LyricWiki/Special_LinksToRedirects.php [hunk 7]
```diff
@@ -133,115 +133,144 @@ function execute(){
...
+					$ids = array_unique($ids);
+
+					$result = $dbr->select(
+						array( $TABLE_PREFIX."page" ),
+						array( 'page_id', 'page_namespace', 'page_title'),
+						array( "page_id IN (".implode(",", $ids).")" ),
+						__METHOD__
+					);
+					if($dbr->numRows($result) > 0){
+						for($cnt=0; $cnt < $dbr->numRows($result); $cnt++){
```
Comment (macbre) [inline extensions/3rdparty/LyricWiki/Special_LinksToRedirects.php]: Please use this instead:

```php
while ($row = $dbr->fetchRow( $result )) {
  // ...
}
```

**Comment by macbre** [comment]
@SeanColombo, ping :)

**Comment by macbre** [comment]
@SeanColombo - ping :)

**Comment by tomasznapieralski** [comment]
Stale PR, closing it.

**Comment by macbre** [comment]
@tomasznapieralski / @SeanColombo  - I think this change is still important although the PR itself is kind of old. `mysql_query` no longer exists in PHP7 - it was removed in favor of `mysqli`. Some piece of Lyrics code simply does not work now.

**Pull request closed without merging**